tests/org.eclipse.reddeer.eclipse.test/src/org/eclipse/reddeer/eclipse/test/jdt/ui/wizards/InterfaceWizardTest.java

```diff
@@ -0,0 +1,178 @@
+import org.eclipse.reddeer.requirements.openperspective.OpenPerspectiveRequirement.OpenPerspective;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Test class for Java Interface Wizard
+ * 
+ * @author zcervink@redhat.com
+ * 
+ */
+@RunWith(RedDeerSuite.class)
+@OpenPerspective(JavaPerspective.class)
+public class InterfaceWizardTest {
+
+	protected static final String PROJECT_NAME = "testproject";
+	protected static final String INTERFACE_NAME = "InterfaceName";
+	protected static final String SOURCE_FOLDER_NAME = PROJECT_NAME + "/src";
+	protected static final String PACKAGE = "org.example.test";
+	protected static final String ENCLOSING_TYPE_CUSTOM_VALUE = "org.example.test." + INTERFACE_NAME;
+
+	@BeforeClass
+	public static void setup() {
+		createTestProject();
+	}
+
+	@Test
+	public void testInterfaceCreatedSuccessfully() {
+		// open and fill the wizard
+		NewInterfaceCreationWizard interfaceDialog = new NewInterfaceCreationWizard();
+		interfaceDialog.open();
+		NewInterfaceWizardPage interfacePage = new NewInterfaceWizardPage(interfaceDialog);
+		interfacePage.setName(INTERFACE_NAME);
+		interfacePage.setPackage(PACKAGE);
+		interfacePage.setSourceFolder(SOURCE_FOLDER_NAME);
+
+		// test toggling the "Enclosing type:" checkbox and changing the default value
+		interfacePage.toggleEnclosingTypeCheckBox(true);
+
+		String enclosingTypeDefaultValue = interfacePage.getEnclosingType();
+		interfacePage.setEnclosingType(ENCLOSING_TYPE_CUSTOM_VALUE);
+		assertTrue(
+				"Interface enclosing type optained from the wizard using getEnclosingType() is not the same as the type given using setEnclosingType().",
+				interfacePage.getEnclosingType().equals(ENCLOSING_TYPE_CUSTOM_VALUE));
+		interfacePage.setEnclosingType(enclosingTypeDefaultValue);
+
+		interfacePage.toggleEnclosingTypeCheckBox(false);
+
+		// test adding, verifying and deleting extended interfaces
+		List<String> extendedIterfaceList = new ArrayList<String>();
+		extendedIterfaceList.add("javax.accessibility.AccessibleAction");
+		extendedIterfaceList.add("javax.accessibility.AccessibleEditableText");
+		extendedIterfaceList.add("javax.accessibility.AccessibleExtendedTable");
+		extendedIterfaceList.add("javax.accessibility.AccessibleStreamable");
+		extendedIterfaceList.add("javax.accessibility.AccessibleText");
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.addExtendedInterface(extendedInterfaceName);
+		}
+
+		assertTrue(
+				"List of interfaces optained using function getExtendedInterfaces() has to be the same as list interfaces, that has been set.",
+				interfacePage.getExtendedInterfaces().equals(extendedIterfaceList));
+
+		for (String extendedInterfaceName : extendedIterfaceList) {
+			interfacePage.removeExtendedInterface(extendedInterfaceName);
+		}
+
+		// toggle generate comments checkbox
+		interfacePage.toggleGenerateComments(true);
```

Comment:
Where is some assert that will verify toggling the box?

Reply:
Added.